### PR TITLE
feat: Pagination token

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
https://github.com/ant-design/ant-design/issues/41884

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Pagination token    |
| 🇨🇳 Chinese |    Pagination token       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8c6addb</samp>

This pull request refactors the pagination component style to use a new `ComponentToken` interface that allows users to customize the theme more easily. It also adds a debug demo and documentation for the component token feature in both Chinese and English.

### <samp>🤖 Generated by Copilot at 8c6addb</samp>

*  Add the component token debug demo for the Pagination component ([link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-c43546e74a8c2619d743e12a158c7e31f84713b197eebf759410ce75abd1934dR1-R7), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-c4073257a836392e88fef00479008a79959c6c41ad6ffbad746bdb90bf0a9530R1-R30), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-0ec69b1c64ecf0246c7bf67d0b562ba1469e3ff049e622ee9be8f4050550d196R30), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-60c09bbacd878424ae64e0cc1b36fbb7c5f5bad7e5bc6a715368690baf2666c7R31))
* Refactor the PaginationToken interface to extract the ComponentToken interface that contains the customizable properties ([link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L12-R34))
* Update the style functions to use the ComponentToken properties instead of the PaginationToken properties ([link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L86-R96), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L142-R153), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L165-R171), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L188-R192), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L195-R217), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L230-R237), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L244-R248), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L252-R255), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L259-R264), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L323-R328), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L366-R373), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L423-R427), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L465-R472), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L482-R485), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L507-R511), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L551-R556), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L626-R629), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L648-R655), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L658-R661), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L671-R679), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L692-R730))
* Update the genComponentStyleHook function to accept a function that returns the ComponentToken based on the token ([link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-ddac945eb46552285af334850cebb6d4ab21a6d9a6b603b3869334c9cc228c66L692-R730))
* Add the table that maps the Less variables to the ComponentToken for the Pagination component in the documentation for migrating Less variables ([link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL114-R130), [link](https://github.com/ant-design/ant-design/pull/42330/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L114-R130))

### CODE REVIEW

File: docs/react/migrate-less-variables.zh-CN.md [hunk 1]
```diff
@@ -201,7 +201,22 @@ title: Less 变量迁移 Design Token
...
+### Pagination 分页
+
+<!-- prettier-ignore -->
+| Less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@pagination-item-bg` | `itemBg` | - |
+| `@pagination-item-size` | `itemSize` | - |
+| `@pagination-item-size-sm` | `itemSizeSM` | - |
+| `@pagination-font-family` | `fontFamily` | 全局 Token |
+| `@pagination-font-weight-active` | `fontWeightActive` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:213]:
```suggestion
| `@pagination-font-weight-active` | `fontWeightStrong` | 全局 Token |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 2]
```diff
@@ -201,7 +201,22 @@ title: Less 变量迁移 Design Token
...
+| `@pagination-item-bg-active` | `itemBgActive` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:214]:
```suggestion
| `@pagination-item-bg-active` | `itemActiveBg` | - |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 3]
```diff
@@ -201,7 +201,22 @@ title: Less 变量迁移 Design Token
...
+| `@pagination-item-link-bg` | `itemLinkBg` | - |
+| `@pagination-item-disabled-color-active` | `itemDisabledColorActive` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:216]:
```suggestion
| `@pagination-item-disabled-color-active` | `itemActiveColorDisabled` | - |
```

File: components/pagination/style/index.tsx [hunk 4]
```diff
@@ -689,34 +692,39 @@ const genBorderedStyle: GenerateStyle<PaginationToken> = (token) => {
...
+    return [
+      genPaginationStyle(paginationToken),
+      token.wireframe && genBorderedStyle(paginationToken),
+    ];
+  },
+  (token) => ({
+    itemBg: token.colorBgContainer,
+    itemSize: token.controlHeight,
+    itemSizeSM: token.controlHeightSM,
+    fontFamily: token.fontFamily,
```
Comment (MadCcc) [inline components/pagination/style/index.tsx:721]: 全局 token 不需要在这里赋值

File: components/pagination/demo/component-token.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+};
+const App: React.FC = () => (
+  <ConfigProvider
+    theme={{
+      components: {
+        Pagination: {
+          itemSize: 20,
+          itemSizeSM: 12,
+          fontFamily: 'SimSun, sans-serif',
+          fontWeightActive: 500,
```
Comment (MadCcc) [inline components/pagination/demo/component-token.tsx:22]: lint

File: components/pagination/style/index.tsx [hunk 6]
```diff
@@ -9,26 +9,27 @@ import { genFocusOutline, genFocusStyle, resetComponent } from '../../style';
 import type { FullToken, GenerateStyle } from '../../theme/internal';
 import { genComponentStyleHook, mergeToken } from '../../theme/internal';
 
+export interface ComponentToken {
+  itemBg: string;
+  itemSize: number;
+  itemActiveBg: string;
+  itemSizeSM: number;
+  itemLinkBg: string;
+  itemDisabledBgActive: string;
```
Comment (MadCcc) [inline components/pagination/style/index.tsx:18]:
```suggestion
  itemActiveBgDisabled: string;
```

Comment (heiyu4585) [reply components/pagination/style/index.tsx]: done

File: docs/react/migrate-less-variables.zh-CN.md (hunk 2 above)
Comment (MadCcc) [reply docs/react/migrate-less-variables.zh-CN.md:214]: 这个还没改

File: components/pagination/demo/component-token.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  return originalElement;
+};
+const App: React.FC = () => (
+  <ConfigProvider
+    theme={{
+      components: {
+        Pagination: {
+          itemSize: 20,
+          itemSizeSM: 12,
+          fontFamily: 'SimSun, sans-serif',
```
[the remark this answers is not in the document]
Comment (heiyu4585) [reply components/pagination/demo/component-token.tsx]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**